graphql-ballerina/tests/09_resources_returning_arrays.bal

```diff
@@ -222,3 +233,51 @@ isolated function testOptionalArrayInvalidQuery() returns error? {
     };
     assertJsonValuesWithOrder(result, expectedPayload);
 }
+
+@test:Config {
+    groups: ["array", "service"]
+}
+isolated function testResourceReturningServiceArrayObjectWithFragmentReturnsError() returns error? {
+    string graphqlUrl = "http://localhost:9100/graphql";
+    string document = string
+    `{
+        allVehicles {
+            ...Details
+        }
+    }
+    fragment Details on Vehicle {
+        name,
+        registeredYear
+    }`;
+    json result = check getJsonPayloadFromService(graphqlUrl, document);
+    json expectedPayload = {
+        data: {
+            allVehicles: [
+                {
+                    name: "Benz",
+                    registeredYear: "2005"
+                },
+                {
+                    name: "BMW",
+                    registeredYear: "2010"
+                },
+                {
+                    name: "Ford"
+                }
+            ]
+        },
+        errors: [
+            {
+                message: "Registered Year is Not Found",
+                locations: [
+                    {
+                        line: 8,
+                        column: 9
+                    }
+                ],
+                path: ["allVehicles", 2, "registeredYear"]
+            }
+        ]
+    };
```

Comment:
This expected value is wrong. The `errors` fields should come before the `data` field. Can you please check why this is not happening here?
If `data` is serialized before the `errors` field, can you please create an issue for that?